src/optionals/process.c

```diff
@@ -268,6 +277,58 @@ static Value runNative(DictuVM* vm, int argCount, Value* args) {
     return execute(vm, argList, true);
 }
 
+#ifdef _WIN32
+static Value killProcess(DictuVM* vm, int argCount, Value* args) {
+    if (argCount > 2) {
+        runtimeError(vm, "kill() takes 1 argument (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_NUMBER(args[0])) {
+        runtimeError(vm, "Argument passed to kill() must be a number");
+        return EMPTY_VAL;
+    }
+
+    pid_t pid = (pid_t)AS_NUMBER(args[0]);
+
+    HANDLE handle = OpenProcess(PROCESS_TERMINATE, TRUE, (int)pid);
+    if (handle != NULL) {   
+        TerminateProcess(handle, 0);
+        CloseHandle(handle);
+    }
+
+    return newResultSuccess(vm, NIL_VAL);
+}
+#else
+static Value killProcess(DictuVM* vm, int argCount, Value* args) {
+    if (argCount > 2) {
+        runtimeError(vm, "kill() takes 1 or 2 arguments (%d given)", argCount);
+        return EMPTY_VAL;
+    }
+
+    if (!IS_NUMBER(args[0])) {
+        runtimeError(vm, "First argument passed to kill() must be a number");
+        return EMPTY_VAL;
+    }
+
+    pid_t pid = (pid_t)AS_NUMBER(args[0]);
+    int signal = 9;
+
+    if (argCount == 2) {
+        if (!IS_NUMBER(args[1])) {
+            runtimeError(vm, "Second argument passed to kill() must be a number");
+            return EMPTY_VAL;
+        }
+
+        signal = AS_NUMBER(args[1]);
+    }
+
+    kill(pid, signal);
```

Comment:
Here we could do:

```suggestion
    if (kill(pid, signal) == -1) {
        ERROR_RESULT;
    }
```

Also means we can pass 0 as the signal and check if (child) processes are still running, which is a way we could test this at least on *nix.

Will have a big of a research about windows